### DISCUSSION TITLE
Multi-domain setup, redirects back to NEXTAUTH_URL

### DISCUSSION THREAD

**Discussion by impact-erik** [nextauthjs/next-auth#5534] [Help]
### Question 💬

I am trying to integrate with Okta using next-auth. I have a multi-domain setup using next-i18next, so I have different urls for any given locale. My NEXTAUTH_URL is configured to be my primary domain, because I have to fill in something 😅.

I am passing the redirect uri to Okta to return to the right domain.
```
signIn('okta', undefined, {redirect_uri: window.location.origin + '/api/auth/callback/okta'})
```

When I try to sign in, I get an error: 
```
[next-auth][error][OAUTH_CALLBACK_ERROR] 
https://next-auth.js.org/errors#oauth_callback_error invalid_grant (The 'redirect_uri' does not match the redirection URI used in the authorization request.) {
  error: OPError: invalid_grant (The 'redirect_uri' does not match the redirection URI used in the authorization request.)
      at processResponse (/Users/erikslagter/Projects/nextauthjs-multidomain/node_modules/openid-client/lib/helpers/process_response.js:38:13)
```

I am curious if this there's a way to have more fine-grained control over how the error is handled? I don't want to fallback to the domain / url specified in `NEXTAUTH_URL`. And also why is there a different in the redirect_uri in the authorisation request?

### How to reproduce ☕️

Clone https://github.com/impact-erik/nextauthjs-multidomain

Add to your host file:
`127.0.0.1 example.en.local example.de.local`

Run the application`yarn dev` 

Click the sign in button in the header. This sends a param `redirect_uri: http://example.en.local:3000/api/auth/callback/okta`

**Issue:**
You are redirected back to: `http://localhost:3000/api/auth/signin?error=OAuthCallback`

### Contributing 🙌🏽

Yes, I am willing to help answer this question in a PR

**Comment by balazsorban44** [comment]
NextAuth.js has its own handling for `redirect_uri` via the redirect callback. Check out the docs: https://next-auth.js.org/configuration/callbacks#redirect-callback

Also, make sure that you allowed the other domain in Okta.

**Reply by impact-erik** [reply]
Hi 👋 @balazsorban44,

I have allowed both domains in Okta. 
I just changed the redirect callback, to not fallback to the baseUrl, but what I see is that in provider.callbackUrl - https://github.com/nextauthjs/next-auth/blob/main/packages/next-auth/src/core/lib/oauth/callback.ts#L114 the `${process.env.NEXTAUTH_URL}/api/auth/callback/okta` is used still/again. 

```
async redirect({ url, baseUrl }) {
      let result = url;
      if (url.startsWith('/')) {
        result = `${baseUrl}${url}`;
      } 
      return result;
    },
```

So I am not 100% if I and how I can override this?!

**Reply by impact-erik** [reply]
My colleague presented https://github.com/nextauthjs/next-auth/issues/600 to me and this resolves my issue
